src/Form/Form.js

```diff
@@ -295,8 +294,8 @@ class Form {
         // This ensures we are not removing the highlight ourselves when autofilling more than once
         if (e && !e.isTrusted) return
 
-        // If the user has changed the value, we prompt to update the stored data
-        this.shouldPromptToStoreData = true
+        // If the user has changed the value, reset shouldPromptToStoreData to initial value
+        this.resetShouldPromptToStoreData()
```

Comment:
Same as above, we were resetting everything to `true` instead of reading from the feature toggle.